File: WorldFeed/Server/src/WorldFeed/Client/Clients/ExecuteClient.cs
namespace WorldFeed.Client.Clients
{
    using System;
    using System.Threading.Tasks;

    using WorldFeed.Common.Public.Client.Clients;
    using WorldFeed.Common.Public.Client.Requesters;
    using WorldFeed.Common.Public.Models.Interfaces;
    using WorldFeed.Common.Web;

    public class ExecuteClient : IExecuteClient
    {
        private readonly IExecuteRequester executeRequester;

        public ExecuteClient(IExecuteRequester executeRequester)
        {
            this.executeRequester = executeRequester;
        }

        public Task<ITwitterResult<T>> AdvanceRequestAsync<T>(Action<ITwitterRequest> configureRequest)
            where T : class
        {
            return this.executeRequester.RequestAsync<T>(configureRequest);
        }

        public Task<ITwitterResult> AdvanceRequestAsync(Action<ITwitterRequest> configureRequest)
        {
            return this.executeRequester.RequestAsync(configureRequest);
        }

        public Task<ITwitterResult<T>> RequestAsync<T>(Action<ITwitterQuery> configureQuery)
            where T : class
        {
            return this.executeRequester.RequestAsync<T>(configureQuery);
        }

        public Task<ITwitterResult> RequestAsync(Action<ITwitterQuery> configureQuery)
        {
            return this.executeRequester.RequestAsync(configureQuery);
        }

        public Task<ITwitterRequest> PrepareTwitterRequestAsync(Action<ITwitterQuery> configureQuery)
        {
            return this.executeRequester.PrepareTwitterRequestAsync(configureQuery);
        }

        public Task<ITwitterRequest> PrepareTwitterRequestAsync(Action<ITwitterRequest> configureRequest)
        {
            return this.executeRequester.PrepareTwitterRequestAsync(configureRequest);
        }
    }
}

File: Sharebook/Client/src/app/sharebook/Client/Clients/ExecuteClient.ts
import {Inject, Injectable} from "@angular/core";

import {ITwitterResult} from "../../../core/Core/Web/TwitterResult";
import {ITwitterRequest} from "../../../core/Public/Models/Interfaces/ITwitterRequest";
import {ITwitterQuery} from 'src/app/core/Public/Models/Interfaces/ITwitterQuery';
import {IExecuteClient} from "../../../core/Public/Client/Clients/IExecuteClient";
import {IExecuteRequester, IExecuteRequesterToken} from "../../../core/Public/Client/Requesters/IExecuteRequester";

@Injectable({
  providedIn: 'root',
})
export class ExecuteClient implements IExecuteClient {
  private readonly _executeRequester: IExecuteRequester;

  constructor(@Inject(IExecuteRequesterToken) executeRequester: IExecuteRequester) {
    this._executeRequester = executeRequester;
  }

  public advanceRequestAsync<T = any>(configureRequest: (twitterRequest: ITwitterRequest) => void): Promise<ITwitterResult<T>> {
    return this._executeRequester.requestAsync<T>(configureRequest);
  }

  public requestAsync<T = any>(configureQuery: (twitterQuery: ITwitterQuery) => void): Promise<ITwitterResult<T>> {
    return this._executeRequester.requestAsync<T>(configureQuery);
  }

  public prepareTwitterRequestAsync(configureRequestOrQuery: (twitterRequest: ITwitterRequest | ITwitterQuery) => void): Promise<ITwitterRequest> {
    return this._executeRequester.prepareTwitterRequestAsync(configureRequestOrQuery);
  }
}
